command/sidechain/enlisting/enlist_validator.go

```diff
@@ -0,0 +1,133 @@
+package enlisting
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/0xPolygon/polygon-edge/command"
+	"github.com/0xPolygon/polygon-edge/command/helper"
+	"github.com/0xPolygon/polygon-edge/command/polybftsecrets"
+	sidechainHelper "github.com/0xPolygon/polygon-edge/command/sidechain"
+	"github.com/0xPolygon/polygon-edge/consensus/polybft/contractsapi"
+	"github.com/0xPolygon/polygon-edge/contracts"
+	"github.com/0xPolygon/polygon-edge/txrelayer"
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/spf13/cobra"
+	"github.com/umbracle/ethgo"
+)
+
+var (
+	stakeManager      = contracts.ValidatorSetContract
+	whitelistFn       = contractsapi.ChildValidatorSet.Abi.Methods["addToWhitelist"]
+	whitelistEventABI = contractsapi.ChildValidatorSet.Abi.Events["AddedToWhitelist"]
+)
+
+var params enlistParams
+
+func GetCommand() *cobra.Command {
+	registerCmd := &cobra.Command{
+		Use:     "enlist-validator",
+		Short:   "enlist a new validator",
+		PreRunE: runPreRun,
+		RunE:    runCommand,
+	}
+
+	setFlags(registerCmd)
+
+	return registerCmd
+}
+
+func setFlags(cmd *cobra.Command) {
+	cmd.Flags().StringVar(
+		&params.accountDir,
+		polybftsecrets.DataPathFlag,
+		"",
+		polybftsecrets.DataPathFlagDesc,
+	)
+
+	cmd.Flags().StringVar(
+		&params.configPath,
+		polybftsecrets.ConfigFlag,
+		"",
+		polybftsecrets.ConfigFlagDesc,
+	)
+
+	cmd.Flags().StringVar(
+		&params.newValidatorAddress,
+		newValidatorAddressFlag,
+		"",
+		"account address of a possible validator",
+	)
+
+	cmd.MarkFlagsMutuallyExclusive(polybftsecrets.DataPathFlag, polybftsecrets.ConfigFlag)
+	helper.RegisterJSONRPCFlag(cmd)
+}
+
+func runPreRun(cmd *cobra.Command, _ []string) error {
+	params.jsonRPC = helper.GetJSONRPCAddress(cmd)
+
+	return params.validateFlags()
+}
+
+func runCommand(cmd *cobra.Command, _ []string) error {
+	outputter := command.InitializeOutputter(cmd)
+	defer outputter.WriteOutput()
+
+	ownerAccount, err := sidechainHelper.GetAccount(params.accountDir, params.configPath)
+	if err != nil {
+		return fmt.Errorf("enlist validator failed: %w", err)
+	}
+
+	inputAddress := types.StringToAddress(params.newValidatorAddress)
```

Comment:
I would put this directly to
```
encoded, err := whitelistFn.Encode([]interface{}{
		[]types.Address{
   		    types.StringToAddress(params.newValidatorAddress)
		},
	})
```
it easier to understand the code that way, i think